### PR TITLE
Add DLR modelzoo

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java [hunk 1]
```diff
@@ -43,10 +45,12 @@ public void testDlr()
         Criteria<Image, Classifications> criteria =
                 Criteria.builder()
                         .setTypes(Image.class, Classifications.class)
+                        .optApplication(Application.CV.IMAGE_CLASSIFICATION)
+                        .optFilter("layers", "50")
                         .optTranslator(translator)
-                        // TODO use the model from S3
-                        .optModelUrls("file:///Users/leecheng/workspace/python/tvm_model_old")
-                        .optEngine("DLR") // use DLR engine
+                        .optModelName("resnet50_v1")
```
Comment (frankfliu) [inline dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java:51]: We should not need optModelName() for built-in model zoo.

File: dlr/dlr-engine/src/test/resources/mlrepo/model/cv/image_classification/ai/djl/dlr/resnet/metadata.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "license": {
+      "name": "The Apache License, Version 2.0",
+      "url": "https://www.apache.org/licenses/LICENSE-2.0"
+    }
+  },
+  "artifacts": [
+    {
+      "version": "0.0.1",
+      "snapshot": false,
+      "name": "resnet50",
```
Comment (frankfliu) [inline dlr/dlr-engine/src/test/resources/mlrepo/model/cv/image_classification/ai/djl/dlr/resnet/metadata.json:20]: use "resnet50_v1" to avoid optModelName()

File: dlr/dlr-engine/build.gradle [hunk 3]
```diff
@@ -12,6 +12,7 @@ dependencies {
         exclude group: "junit", module: "junit"
     }
     testImplementation "org.slf4j:slf4j-simple:${slf4j_version}"
+    testRuntimeOnly project(":model-zoo")
```
Comment (frankfliu) [inline dlr/dlr-engine/build.gradle:15]: Why we need this?

File: dlr/dlr-engine/src/test/resources/mlrepo/model/cv/image_classification/ai/djl/dlr/resnet/metadata.json (hunk 2 above)
Comment (stu1130) [reply dlr/dlr-engine/src/test/resources/mlrepo/model/cv/image_classification/ai/djl/dlr/resnet/metadata.json]: done

**Review by frankfliu — APPROVED** [review]
